questionpy_sdk/webserver/state_storage.py

```diff
@@ -50,11 +96,13 @@ def _parse_form_element(self, form_element: FormElement, form_data: dict) \
         elif isinstance(form_element, GroupElement):
             group = {}
             for child in form_element.elements:
-                if not isinstance(child, StaticTextElement) and child.name in form_data:
-                    group[child.name] = self._parse_form_element(child, form_data)
+                if not isinstance(child, StaticTextElement) and child.name in form_data[form_element.name]:
+                    group[child.name] = self._parse_form_element(child, form_data[form_element.name])
```

Comment:
Wird eine optionale Gruppe nicht angegeben, fliegt in `_parse_form_element` noch eine Exception. Stattdessen sollten die Form-Daten einfach ohne den Eintrag an das Paket gegeben werden, damit da der Default, den ich kürzlich nachgezogen habe, greifen kann.

Reply:
Ok habe ich so gemacht, indem `_parse_form_element` dann None zurückgibt und in `_parse_section` dann kein Eintrag erstellt wird.

Reply:
Es bleibt noch ein Problem: Wenn in einem Multi-Select nur eine Option ausgewählt wird, wird ein String gesendet, im SDK aber eine Liste erwartet. Moodle fügt den Namen von Multi-Selects `[]` hinzu und entweder PHP selbst oder Moodle macht beim POST daraus dann Listen. Es wäre auch denkbar, im FormModel automatisch einen Validator hinzuzufügen, der eine Liste um den Wert wickelt. Was denkst du dazu?

Reply:
Also soll dann auch wenn `multiple=False` eine Liste um den Wert gewickelt werden? Falls ja, finde ich das gut.

Reply:
Nein, nur bei `multiple=True`.

Reply:
Ok ist gefixt: [508b014](https://github.com/questionpy-org/questionpy-sdk/pull/42/commits/508b014d8624093140a214ff9a157ad472e07c2f)